### PR TITLE
Raptor solution reader: first draft of the reader

### PR DESCRIPTION
**Pull request by TeXitoi**
This pull request is not on dev, but on a feature branch that we will use until its mergeable.

### CODE REVIEW

File: source/routing/raptor.cpp [hunk 1]
```diff
@@ -602,7 +607,9 @@ void RAPTOR::raptor_loop(Visitor visitor, const type::AccessibiliteParams & acce
                             // We want to update the labels, if it's better than the one computed before
                             // Or if it's an destination point if it's equal and not unitialized before
                             const bool best_add_result = this->b_dest.add_best(visitor, jpp.sp_idx, workingDt, this->count);
-                            if (visitor.comp(workingDt, bound) || best_add_result) {
+                            if ((!visitor.comp(bound, workingDt)
+                                 && visitor.comp(workingDt, working_labels.mut_dt_pt(jpp.sp_idx)))
+                                || best_add_result) {
```
Comment (antoine-de) [inline source/routing/raptor.cpp:612]: shouldn't we swap the tests ?

File: source/routing/raptor.cpp [hunk 2]
```diff
@@ -81,9 +81,9 @@ bool RAPTOR::apply_vj_extension(const Visitor& v, const bool global_pruning,
             const auto sp_idx = SpIdx(*sp);
             const DateTime bound = (v.comp(best_labels_pts[sp_idx], b_dest.best_now) || !global_pruning) ?
                                         best_labels_pts[sp_idx] : b_dest.best_now;
-            if(!v.comp(workingDt, bound)) {
-                continue;
-            }
+            if (v.comp(bound, workingDt)) { continue; }
+            if (!v.comp(workingDt, working_labels.dt_pt(sp_idx))) { continue; }
```
Comment (l-vincent-l) [inline source/routing/raptor.cpp:85]: Can you add a comment on that?

File: source/routing/raptor.cpp [hunk 3]
```diff
@@ -286,6 +288,9 @@ RAPTOR::compute_all(const vec_stop_point_duration& departures_,
     }
     BOOST_ASSERT( departures.size() > 0 );    //Assert that reversal search was symetric
 
+    if (nb_sol_reader < result.size()) {
+        std::cout << "***** error reader fail" << nb_sol_reader << " " << result.size() << std::endl;
```
Comment (l-vincent-l) [inline source/routing/raptor.cpp:292]: It should be a log isn't it?

File: source/routing/raptor.cpp (hunk 1 above)
Comment (TeXitoi) [inline source/routing/raptor.cpp:612]: We can, but the goal is to remove b_dest

File: source/routing/raptor.cpp (hunk 2 above)
Comment (TeXitoi) [inline source/routing/raptor.cpp:85]: I don't really know how to do that cleanly as it is done in several different location, here and in apply_vj_extention, and the equivalent is done in foot_path. Adding comments at all these locations seems dangerous, but I do not have any better idea. Do you have?

File: source/routing/raptor.cpp (hunk 3 above)
Comment (TeXitoi) [inline source/routing/raptor.cpp:292]: That's pure debugging for the moment as the reader just do nothing more than printing things.

**Pull request merged**